### ISSUE TITLE
Can't load custom fonts

### ISSUE THREAD

**Issue by belarusaed** [Automattic/node-canvas#2097]
## Issue

Can't load custom fonts that not installed to `/usr/share/fonts/`.

My font: https://fonts.google.com/specimen/Montserrat

It still loads fonts from `/usr/share/fonts`, but can't load fonts with `registerFont`

## Steps to Reproduce
1. Unzip font archive to `Montserrat/` into script `__dirname`
2. Run script:
```js
const path = require('path');
const fs = require('fs');
const { registerFont, createCanvas } = require('canvas');

registerFont(path.join(__dirname, 'Montserrat', 'static', 'Montserrat-Regular.ttf'), { family: 'Montserrat' });

const canvas = createCanvas(500, 500);
const ctx = canvas.getContext('2d');

ctx.font = '48px Montserrat';
ctx.textAlign = 'center';
ctx.fillText('Test text 123', 250, 250);

canvas.createPNGStream().pipe(fs.createWriteStream('image.png'));
```

## Your Environment
* Version of node-canvas (output of `npm list canvas` or `yarn list canvas`): canvas@2.9.3
* Environment (e.g. node 4.2.0 on Mac OS X 10.8): `node v16.14.2`, `Linux fedora 5.18.17-200.fc36.x86_64 #1 SMP PREEMPT_DYNAMIC Thu Aug 11 14:36:06 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux`

**Comment by belarusaed** [comment]
Also, it works normally if I just remove `registerFont` and install font to system. But I need to load font from project directory.

**Comment by virtulis** [comment]
Can confirm the same problem with 2.9.1 to 2.9.3 in multiple Linux environments. Reverting to 2.9.0 helps so probably related to https://github.com/Automattic/node-canvas/pull/1987

**Comment by simonhaenisch** [comment]
I'm just investigating a similar issue on linux (in Docker) where `registerFont` doesn't fail but also doesn't register the provided font, and I could figure out the following facts:

- It only happens when there is any custom font installed, either in `/usr/share/fonts` or in `$HOME/.local/share/fonts`. If both of those directories are empty, then `registerFont` works fine.
- It doesn't happen with `canvas@2.9.0` but any later version (e.g. `2.9.1` or `2.10.2`), so the assumption that it was caused by the changes of #1987 makes a lot of sense.

**Comment by chearon** [comment]
Yeah, #1987 created a new problem. It only worked with no font configuration files on the system. It turns out that the changes to `FcConfig` in that PR get overwritten later by what the OS provides, which creates a worse situation than before since we only set the postscript name field.

Trying to register and select custom fonts with OS APIs is just not going to work, so I'm waiting until Pango 2 is released to revamp the whole thing using their custom font APIs.

**Comment by KevinUK** [comment]
I have the same issue, I'm sure I had it working for a time with Canvas `2.10.2` on Node.js 18 LTS though.

Can you please tell me what version of Node.js you are running for Canvas `2.9.0`? I get compilation errors when I downgrade Canvas.

I'm also using fabric.js so maybe I need to downgrade that at the same time.

"dependencies": {
    "canvas": "2.11.2",
    "fabric": "5.3.0",
    "fontfaceobserver": "2.3.0"
  }

**Comment by ggolda** [comment]
Doesn't work for me either, custom fonts work only if they are installed in `/usr/share/fonts`.

**Comment by mkbaranovskyi** [comment]
I experience this bug on Ubuntu 24.04 with the latest `canvas@2.11.2`. And I can't fallback to `canvas@2.9.0` - it errors on Node 20. 
Thank you @simonhaenisch for the suggested solutions, registering fonts globally worked for me (although you have to be careful with font names: e.g. `AlexBrush-Regular.ttf` it registers as `AlexBrush`, not `AlexBrush-Regular` as one might expect).

**Comment by ilya-adnymics** [comment]
I've found that on Ubuntu 24.04 (canvas@2.11.2, node@20.17.0) I could make it work by setting an environment variable `FONTCONFIG_FILE` with a path to a font config override which excluded certain fonts. Below is the conf file I used (named 78-Reject.conf)
```xml
<?xml version="1.0"?>
<!DOCTYPE fontconfig SYSTEM "urn:fontconfig:fonts.dtd">
<!-- Generated by Font Manager. Do NOT edit this file. -->
<fontconfig>
  <selectfont>
    <rejectfont>
      <pattern>
        <patelt name="family">
          <string>Ubuntu</string>
        </patelt>
      </pattern>
      <pattern>
        <patelt name="family">
          <string>Ubuntu Mono</string>
        </patelt>
      </pattern>
      <pattern>
        <patelt name="family">
          <string>Ubuntu Sans</string>
        </patelt>
      </pattern>
      <pattern>
        <patelt name="family">
          <string>Ubuntu Sans Mono</string>
        </patelt>
      </pattern>
      <pattern>
        <patelt name="family">
          <string>Noto Color Emoji</string>
        </patelt>
      </pattern>
    </rejectfont>
  </selectfont>
</fontconfig>
```

Not sure how it behaves on other operating systems, for me the error was that the `registerFont` simply did nothing - the custom font wasn't applied, but rather some system fallback font. I spent some time figuring out which fallbacks were used (you can see them in conf file above), and after applying font config override, text is rendered using my custom font.
This solution shouldn't require installing custom fonts globally.
